src/global/logger.type.ts

```diff
@@ -0,0 +1,14 @@
+import { ValuesType } from '../types/helpers/valuesType';
+
+export const LogLevelIndex = {
+  DEBUG: 5,
+  INFO: 4,
+  WARN: 3,
+  ERROR: 2,
+  FATAL: 1,
+  OFF: 0,
+};
+
+export type LogLevelIndex = ValuesType<typeof LogLevelIndex>;
+
+export type LogLevel = keyof typeof LogLevelIndex & string & NonNullable<unknown>;
```

Comment:
I think the ` & string & NonNullable<unknown>` segment can be removed.

Reply:
Yep, you are right, it looks like autocomplete with the keyof typeof work without shenanigans.